src/renderer/app.jsx

```diff
@@ -71,32 +117,75 @@ const ScratchDesktopHOC = function (WrappedComponent) {
         }
         render () {
             const shouldShowTelemetryModal = (typeof ipcRenderer.sendSync('getTelemetryDidOptIn') !== 'boolean');
+
+            const childProps = omit(this.props, Object.keys(ScratchDesktopComponent.propTypes));
+
             return (<WrappedComponent
                 canEditTitle
+                canModifyCloudData={false}
                 isScratchDesktop
-                projectId={defaultProjectId}
-                projectTitle={this.state.projectTitle}
                 showTelemetryModal={shouldShowTelemetryModal}
                 onClickLogo={this.handleClickLogo}
                 onProjectTelemetryEvent={this.handleProjectTelemetryEvent}
                 onStorageInit={this.handleStorageInit}
                 onTelemetryModalOptIn={this.handleTelemetryModalOptIn}
                 onTelemetryModalOptOut={this.handleTelemetryModalOptOut}
                 onUpdateProjectTitle={this.handleUpdateProjectTitle}
-                {...this.props}
+
+                // allow passed-in props to override any of the above
+                {...childProps}
             />);
         }
     }
 
-    return ScratchDesktopComponent;
+    ScratchDesktopComponent.propTypes = {
+        loadingState: PropTypes.oneOf(LoadingStates),
+        onFetchedInitialProjectData: PropTypes.func,
+        onHasInitialProject: PropTypes.func,
+        onLoadedProject: PropTypes.func,
+        onLoadingCompleted: PropTypes.func,
+        onLoadingStarted: PropTypes.func,
+        onRequestNewProject: PropTypes.func,
+        vm: PropTypes.instanceOf(VM).isRequired
+    };
+    const mapStateToProps = state => {
+        const loadingState = state.scratchGui.projectState.loadingState;
+        return {
+            loadingState: loadingState,
+            vm: state.scratchGui.vm
+        };
+    };
+    const mapDispatchToProps = dispatch => ({
+        onLoadingStarted: () => dispatch(openLoadingProject()),
+        onLoadingCompleted: () => dispatch(closeLoadingProject()),
+        onHasInitialProject: (hasInitialProject, loadingState) => {
```

Comment:
Maybe this is me being too picky, but it seems like you could refactor things so there is no `onHasInitialProject()` function, and instead you call one mapDispatchToProps function to just call `requestProjectUpload()`, and a separate mapDispatchToProps function to just call `setProjectId()`.

Reply:
Yeah, I wasn't sure what the convention was here. I was struggling to give these functions meaningful names and this one seemed like a natural combination, but if it's better to have `mapDispatchToProps` be a sort of "dispatch curry" and no more then I'm happy to do this refactor.

Reply:
In the interest of time I'm going to go ahead and merge this PR. I filed #154 so I don't forget to address this refactor in a followup change.